tests/test_logic.py

```diff
@@ -147,6 +150,47 @@ def test_change_project_slug(cookiecutter_template_directory: Path,
     assert not project_directory.joinpath(old_project_slug).exists()
 
 
+def test_changes_from_zip_file(cookiecutter_template_directory: Path,
+                               cookies: Cookies):
+    result: Result = cookies.bake(extra_context=SAMPLE_CONTEXT, template=str(cookiecutter_template_directory))
+    if result.exception is not None:
+        raise result.exception
+    project_directory = Path(result.project)
+    _initialize_git_repo(project_directory)
+    subprocess.run(["git", "add", "-A"], cwd=str(project_directory), check=True)
+    subprocess.run(["git", "commit", "-m", "initial"], cwd=str(project_directory), check=True)
+
+    context = json.loads(project_directory.joinpath("docs", "cookiecutter_input.json").read_text(encoding="utf-8"))
+
+    cookiecutter_template_directory.joinpath("{{cookiecutter.project_slug}}",
+                                             "README.rst").write_text("updated readme for zip test")
+    subprocess.run(["git", "add", "-A"], cwd=str(cookiecutter_template_directory), check=True)
+    subprocess.run(["git", "commit", "-m", "updated readme for zip test"],
+                   cwd=str(cookiecutter_template_directory), check=True)
```

Comment:
nice work with the test! as another nit, I can offer you shorthands `subprocess.check_call` which does the above, and `subprocess.check_output` to additionally return the output of the command that was run

Reply:
I think subprocess.run() will be more appropriate here which was added in python 3.5 and is recommended way. Since we running tests with python v3.5+ we should be fine with run() rather than older call. What do you think?